File: determine-if-string-halves-are-alike/determine-if-string-halves-are-alike.java
class Solution {
    public boolean halvesAreAlike(String s) {
        
        String a = s.substring(0, s.length() / 2),
                b = s.substring(s.length() / 2);
        
        return vowelCount(a) == vowelCount(b);
    }
    
    private int vowelCount(String s) {
        
        int count = 0;
        
        for (char c : s.toCharArray()) {
            
            if (isVowel(c)) {++count;}
        }
        
        return count;
    }
    
    private boolean isVowel(char ch) {
        
        return (ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u' || 
                ch == 'A' || ch == 'E' || ch == 'I' || ch == 'O' || ch == 'U');
    }
}

File: determine-if-string-halves-are-alike/determine-if-string-halves-are-alike.cpp
class Solution {
public:
    bool halvesAreAlike(string& s) {
        
        int l1 = s.length(), l2 = l1 >> 1;
        short int count = 0;
        
        for (int i = 0; i < l2; ++i) {
            
            if (isVowel(s[i])) {
                
                ++count;
            }
        }
        
        for (int j = l2; j < l1; ++j) {
            
            if (isVowel(s[j])) {
                
                --count;
            }
        }
        
        return count == 0;
    }
    
private:
    bool isVowel(char& ch) {
        
        return (ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u' || 
                ch == 'A' || ch == 'E' || ch == 'I' || ch == 'O' || ch == 'U');
    }
};